service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredTaskExecutor.java

```diff
@@ -79,8 +90,26 @@ public void execute(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
             messageContext.resetMessageContext();
         }
 
+        var payload = hl7TranslatedResponse;
+        if (isLargeEhrExtract(hl7TranslatedResponse)) {
+            // TODO: 16/08/2021
+            String compressedHl7 = hl7TranslatedResponse;
```

Comment:
I think you can just leave the TODO without any code related to compressing as even a compressed ehrExtract can still be too large